module/os/freebsd/zfs/zvol_os.c

```diff
@@ -146,8 +147,19 @@ SYSCTL_NODE(_vfs_zfs, OID_AUTO, vol, CTLFLAG_RW, 0, "ZFS VOLUME");
 SYSCTL_INT(_vfs_zfs_vol, OID_AUTO, mode, CTLFLAG_RWTUN, &zvol_volmode, 0,
 	"Expose as GEOM providers (1), device files (2) or neither");
 static boolean_t zpool_on_zvol = B_FALSE;
+static unsigned int zvol_threads = 0;
+static unsigned int zvol_num_taskqs = 0;
+static unsigned int zvol_request_sync = 1;
 SYSCTL_INT(_vfs_zfs_vol, OID_AUTO, recursive, CTLFLAG_RWTUN, &zpool_on_zvol, 0,
 	"Allow zpools to use zvols as vdevs (DANGEROUS)");
+SYSCTL_UINT(_vfs_zfs_vol, OID_AUTO, threads, CTLFLAG_RWTUN, &zvol_threads, 0,
+	"Number of threads for I/O requests. Set to 0 to use all active CPUs");
+SYSCTL_UINT(_vfs_zfs_vol, OID_AUTO, taskqs, CTLFLAG_RWTUN, &zvol_num_taskqs, 0,
+	"Number of zvol taskqs");
+SYSCTL_UINT(_vfs_zfs_vol, OID_AUTO, sync, CTLFLAG_RWTUN, &zvol_request_sync, 0,
+	"Synchronously handle bio requests");
```

Comment:
If you are making the code OS-independent, why are the tunables still OS-specific?

Reply:
Variables definitions are moved to common code (zvol.c), also declared in zvol _impl.h

To be honest, I am not sure about clean way, how to declare sysctl on FreeBSD side same as module param on linux side. Am I right, that I should use ZFS_MODULE_PARAM macros and define them in zvol.c?

Reply:
Right.

Reply:
I've defined `zvol_threads`, `zvol_num_taskqs` and `zvol_request_sync` module params in the zvol.c and got the zvol sysctls naming inconsistency on FreeBSD side:
```
root@fb:~ # sysctl -a | grep vol
vfs.zfs.zvol_request_sync: 0
vfs.zfs.zvol_num_taskqs: 0
vfs.zfs.zvol_threads: 0
vfs.zfs.zvol_enforce_quotas: 1
vfs.zfs.vol.unmap_enabled: 1
vfs.zfs.vol.recursive: 0
vfs.zfs.vol.mode: 1
```

It could be seen, that part of parameters is declared with `.zvol_` prefix and other with `vol` prefix.
From other side, the module parameters names are unified with Linux. Later, it is possible to rename/unify `.vol` prefix sysctls, if it will not harmful anything.